src/consensus/params.h

```diff
@@ -29,6 +29,11 @@ struct BIP9Deployment {
     int64_t nStartTime;
     /** Timeout/expiry MedianTime for the deployment attempt. */
     int64_t nTimeout;
+    /** If lock in occurs, delay activation until at least this block
+     *  height.  Note that activation will only occur on a retarget
+     *  boundary.
+     */
+    int min_activation_height{0};
```

Comment:
It's a **minimum** activation height, because activation could be later than that if min_activation_height is reached prior to  timeout MTP being reached. With a fast enough increase in hashrate, you could probably reach any reasonable min_activation_height prior to timeout MTP being reached; and of course min_activation_height could be deliberately set in prior to or in the middle of the signalling period.

minimum activation height is a fixed value, not a delta; so for your example, if client version X specifies min_activation_height H1 and version Y specifies min_activation_height H2 with H2<H1 then you'd expect Y to be enforcing different rules between H2..H1, because that's what you've told it to do. The only time they'd agree on the rules is if lockin occurs at a height >= H1-2016.

Any change to deployment parameters can cause a change to what a client considers to be a valid block, and cause a consensus split; this isn't any different to changing the bit, starttime, timeout, period or threshold...

Reply:
> It's a minimum activation height, because activation could be later than that if min_activation_height is reached prior to timeout MTP being reached. With a fast enough increase in hashrate, you could probably reach any reasonable min_activation_height prior to timeout MTP being reached; and of course min_activation_height could be deliberately set in prior to or in the middle of the signalling period.

Right, that said if you want to achieve ST goal's of backend delay, you should have a retarget period buffer between expected height of timeout and min_activation_height to swallow even a doubling of hashrate, or any other kind of scaling factor we deem *physically* unlikely to happen. Though this conversation belongs more to activation parameters selection. I agree, this logic could serve beyond ST-style of deployment.

> minimum activation height is a fixed value, not a delta; so for your example, if client version X specifies min_activation_height H1 and version Y specifies min_activation_height H2 with H2<H1 then you'd expect Y to be enforcing different rules between H2..H1, because that's what you've told it to do. The only time they'd agree on the rules is if lockin occurs at a height >= H1-2016.

I'm just concerned about successfully LOCKED_IN then having uasf folks bothered by the backend delay and willingly to fasten activation by tweaking `min_activation_height`. I'm not sure the immutable semantic of `min_activation_height` is that much understood. Though there is a thousand ways to fork yourself of the network if you have the intent, we can't prevent all of them to happen...

>  Any change to deployment parameters can cause a change to what a client considers to be a valid block, and cause a consensus split; this isn't any different to changing the bit, starttime, timeout, period or     threshold...

Yes, but I think "Once activation parameters have been released for a consensus change, they MUST NOT be changed to avoid risks of consensus split until reaching a final deployment state" on top `BIP9Deployment` is good practice. Future code reviewers might not be as much familiar than us w.r.t to those consensus subtleties.

Reply:
> I'm just concerned about successfully LOCKED_IN then having uasf folks bothered by the backend delay and willingly to fasten activation by tweaking min_activation_height

I don't think there's any community concerns with the backend delay. Making sure economic majority activates is in everyone's interests.

Reply:
I think this is also desirable in the future in certain circumstances, and unlikely to be an issue for ST.

1) For ST: The time is *really short*. Backing it up after lock in would be unfair to smaller miners (< 10% total hash) who need more time to upgrade, so it would run counter to norms...
2) In the future: it would be possible to do a Long-Range-Activation, e.g. a ST but with min_active_height far in the future (e.g., 2 years). Once it is locked in, then developers can start working on broader ecosystem support & wallet software integration knowing when it will be available. Once that seems ready, it is a soft-fork to decrease the min_height in a future release (but it still should be signalled for, but even if it isn't decreasing it is just a flag day soft fork, not hard fork AFAIU). I think that sounds like too much unnecessary fuss, but I _could_ imagine it being used.